File: go-sdk/pkg/models/shared/createroom.go
package shared

type CreateRoom struct {
	DefaultAccesses []string               `json:"defaultAccesses"`
	GroupsAccesses  map[string]interface{} `json:"groupsAccesses,omitempty"`
	ID              string                 `json:"id"`
	Metadata        map[string]interface{} `json:"metadata,omitempty"`
	UsersAccesses   map[string]interface{} `json:"usersAccesses,omitempty"`
}

File: typescript-sdk/src/sdk/models/shared/createroom.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class CreateRoom extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=defaultAccesses" })
  defaultAccesses: string[];

  @SpeakeasyMetadata({ data: "json, name=groupsAccesses" })
  groupsAccesses?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=metadata" })
  metadata?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=usersAccesses" })
  usersAccesses?: Record<string, any>;
}
